Intersect (Core)/Enums/GameObjectTypeExtensions.cs

```diff
@@ -54,5 +55,111 @@ public static IDatabaseObject CreateNew(this GameObjectType gameObjectType)
+        }
+
+        public static VariableDataType GetVariableType(this GameObjectType gameObjectType, Guid variableDescriptorId)
+        {
+            var lookup = gameObjectType.GetLookup();
+
+            try
+            {
+                return lookup.ValueList
+                    .Cast<IVariableBase>()
+                    .FirstOrDefault(var => var.Id == variableDescriptorId)?.Type ?? 0;
+            }
+            catch (InvalidCastException e)
+            {
+                Logging.Log.Error($"Invalid cast! Tried to get a variable value for a GameObject that is not represented by a class inheriting IVariableBase!");
+                return 0;
+            }
+        }
+
+        public static Guid IdFromList(this GameObjectType gameObjectType, int listIndex, VariableDataType dataTypeFilter = 0)
+        {
+            var lookup = gameObjectType.GetLookup();
+
+            if (listIndex < 0 || listIndex > lookup.KeyList.Count)
+            {
+                return Guid.Empty;
+            }
+
+            if (dataTypeFilter == 0)
+            {
+                return lookup.KeyList.OrderBy(pairs => lookup[pairs]?.Name).ToArray()[listIndex];
+            }
+
+            try
+            {
+                return lookup
+                    .OrderBy(kv => kv.Value?.Name)
+                    .Select(kv => kv.Value)
+                    .Cast<IVariableBase>()
+                    .Where(desc => desc.Type == dataTypeFilter)
+                    .Select(desc => desc.Id)
+                    .ToArray()[listIndex];
+            }
+            catch (InvalidCastException e)
+            {
+                Logging.Log.Error($"Invalid cast! A VariableDataFilter was probably passed for a game object that does not inherit IVariableBase!");
+                return Guid.Empty;
+            }
+        }
+
+        public static string[] Names(this GameObjectType gameObjectType, VariableDataType dataTypeFilter = 0) 
+        {
+            if (dataTypeFilter == 0)
+            {
+                return gameObjectType
+                    .GetLookup()
+                    .OrderBy(p => p.Value?.Name)
+                    .Select(pair => pair.Value?.Name ?? Deleted)
+                    .ToArray();
+            }
+
+            try
+            {
+                return gameObjectType
+                    .GetLookup()
+                    .Select(kv => kv.Value)
+                    .Cast<IVariableBase>()
+                    .Where(desc => desc.Type == dataTypeFilter)
+                    .OrderBy(p => p?.Name)
+                    .Select(pair => pair?.Name ?? Deleted)
+                    .ToArray();
+            }
+            catch (InvalidCastException e)
+            {
+                Logging.Log.Error($"Invalid cast! A VariableDataFilter was probably passed for a game object that does not inherit IVariableBase!");
+                return new string[0];
+            }
+        } 
+
+        public const string Deleted = "ERR_DELETED";
```

Comment:
Reused DatabaseObject's copy of this?

Reply:
I can't find a good way of re-using that. I don't have access to a `DatabaseObject` at the points I'm needing this string